src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/AnalysisContextExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class AnalysisContextExtensions
+    {
+        /// <summary>
+        /// Returns true if the given <paramref name="span"/> is in the analysis span of the given <paramref name="context"/>,
+        /// i.e. either of the following is true:
+        ///  - <see cref="SyntaxTreeAnalysisContext.FilterSpan"/> is <code>null</code> (we are analyzing the entire tree)
+        ///  OR
+        ///  - <paramref name="span"/> intersects with <see cref="SyntaxTreeAnalysisContext.FilterSpan"/>.
+        /// </summary>
+        public static bool IsInAnalysisSpan(this SyntaxTreeAnalysisContext context, TextSpan span)
+            => !context.FilterSpan.HasValue || span.IntersectsWith(context.FilterSpan.Value);
```

Comment:
nit: `ShouldAnalyzeSpan` seems like a reasonable name.  i personally find `in` something that has overloaded meaning and makes me wary about what it means.